### DISCUSSION TITLE
How were payload injection offsets determined?

### DISCUSSION THREAD

**Discussion by tsujp** [asmvik/yabai#1288] [Q&A] [answered]
Hello once again @koekeishiya!

I've been scouring the Issues tab for about an hour now and I cannot find the thread I (think I) saw a few months ago where someone asked how you figured out the offsets and what not for the scripting payload injection into Dock.app. You mentioned someone having a good guide and linked it too I believe.

I'm looking to do some hacking on Yabai now and I am wondering if that thread exists (I'm very sure it does but I cannot find it) and was wondering if you had some material or guidance on reverse engineering these things to figure out what you have with regards to injecting code into macOS processes, and Dock.app.

**Comment by tsujp** [comment, marked as answer]
Oh my goodness the _second_ I posted this I realised I didn't do a search for `reverse offset` on the Issues tab, and sure enough I found it!

https://github.com/koekeishiya/yabai/issues/971

**Reply by asmvik** [reply]
I don't think I really have more notes than what is written in that issue.

**Reply by tsujp** [reply]
Roger. I'll leave this up and/or suggest you do too if only for more search terms to match against for others who may hunt for the same information as I was?

**Reply by tsujp** [reply]
> I don't think I really have more notes than what is written in that issue.

Also, a thank you for the explanation too on that Issue thread by the way :)
